File: sorting/insertion_sort.py
from typing import List, Union
def insertion_sort(a: List[Union[int, float]], n: int) -> List[Union[int, float]]:
    
    for i in range(1, n):
        pivot = a[i]
        j = i - 1

        while j >= 0 and a[j] > pivot: 
            a[j+1] = a[j]
            j -= 1
        
        a[j+1] = pivot 
    
    return a












def insertion_sort2(a: List[Union[int, float]], n: int) -> List[Union[int, float]]:

    for i in range(1, n):

        pivot = a[i]
        j = i - 1

        while j>= 0 and a[j] > pivot:
            a[j+1] = a[j]
            j -= 1
        
        a[j+1] = pivot 
    
    return a 






















if __name__ == '__main__':
    array = [1, 5, 2, 10, 2, 3, 4, 7, 9]
    print(insertion_sort2(a=array, n=len(array)))

File: sorting/insertion_sort.c
#include<stdio.h>

#define TAM 10


void insertion_sort(int a[], int n){
    int pivot, j;

    for(int i = 1; i < n; i++){
        pivot = a[i];
        j = i - 1;

        while(j >= 0 && a[j] > pivot){
            a[j+1] = a[j];
            j--;
        }
        a[j + 1] = pivot;
    }

}

void print_array(int a[], int n){
    for(int i = 0; i < n; i++){
        printf("%d\n", a[i]);
    }
}


int main(void){
    int a[TAM] = {5, 3 ,1, 7, 6, 2, 4, 10, 8 ,9};
    int pivot, j;
    print_array(a, TAM);
    insertion_sort(a, TAM);
    printf("\n-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n");
    print_array(a, TAM);

}